File: backend/app/Models/WorkShift.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class WorkShift extends Model
{
    use HasFactory;

    protected $fillable = [
        'shift_name',
        'start_time',
        'end_time',
        'break_time',
    ];

    public function workSchedules(): HasMany
    {
        return $this->hasMany(WorkSchedule::class, 'shift_id');
    }
}

File: hr_management_app/lib/models/work_shift.dart
class WorkShift {
  final String shiftName;
  final String startTime;
  final String endTime;
  final int breakTime;
  final String createdAt;
  final String updatedAt;

  WorkShift({
    required this.shiftName,
    required this.startTime,
    required this.endTime,
    required this.breakTime,
    required this.createdAt,
    required this.updatedAt,
  });

  factory WorkShift.fromJson(Map<String, dynamic> json) {
    return WorkShift(
      shiftName: json['shift_name'] ?? '',
      startTime: json['start_time'] ?? '',
      endTime: json['end_time'] ?? '',
      breakTime: json['break_time'] ?? 0,
      createdAt: json['created_at'] ?? '',
      updatedAt: json['updated_at'] ?? '',
    );
  }
}